portfolio-project-react/src/components/SkillItem/SkillItem.jsx

```diff
@@ -0,0 +1,65 @@
+import React, { useEffect, useState } from "react";
+import styles from "./SkillItem.module.css";
+import { CHAR_UNIT, DELETE, PLACEHOLDER_ADD } from "./constants";
+
+const SkillItem = ({ isEdit, data, index, isAddNewItem, onAdd, onDelete }) => {
+  const [skill, setSkill] = useState(data);
+  const [newSkill, setNewSkill] = useState("");
+  const handleEvent = (e) => {
+    if (e.code === "Enter" && newSkill.length > 0) {
+      onAdd(newSkill);
+      setNewSkill("");
+    }
+  };
+  const [width, setWidth] = useState(
+    skill ? skill.length + 1 + CHAR_UNIT : 1 + CHAR_UNIT
+  );
```

Comment:
if u want to initialize the useState conditionally, wrap the condition logic in a callback fn.